presto-main/src/main/java/com/facebook/presto/cost/UnnestStatsRule.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.ComposableStatsCalculator.Rule;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.UnnestNode;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.planner.plan.Patterns.unnest;
+
+public class UnnestStatsRule
+        implements Rule<UnnestNode>
+{
+    private static final int UPPER_BOUND_ROW_COUNT_FOR_ESTIMATION = 1;
+
+    @Override
+    public Pattern<UnnestNode> getPattern()
+    {
+        return unnest();
+    }
+
+    @Override
+    public Optional<PlanNodeStatsEstimate> calculate(UnnestNode node, StatsProvider statsProvider, Lookup lookup, Session session, TypeProvider types)
+    {
+        PlanNodeStatsEstimate sourceStats = statsProvider.getStats(node.getSource());
+        PlanNodeStatsEstimate.Builder calculatedStats = PlanNodeStatsEstimate.builder();
+        if (sourceStats.getOutputRowCount() > UPPER_BOUND_ROW_COUNT_FOR_ESTIMATION) {
```

Comment:
Personally i would make it even stricter. I would only enable it if the number of columns is `1`, otherwise we are miss-estimating size of the other columns (it is correct that the size of a single ARRAY remains the same after unnest, but in case of any other column in the input set - the total size of the input column must be multiplied by the UNNEST factor).

Reply:
It's actually not whether the number of columns is one. It's whether the original column is kept. So `node.getReplicateSymbols().isEmpty()` is good enough?

Reply:
Yeah, i think it is fair.